lib/vquic/curl_ngtcp2.c

```diff
@@ -246,10 +246,37 @@ static void h3_data_done(struct Curl_cfilter *cf, struct Curl_easy *data)
    the maximum packet burst to MAX_PKT_BURST packets. */
 #define MAX_PKT_BURST 10
 
+struct pkt_io_ctx {
+  struct Curl_cfilter *cf;
+  struct Curl_easy *data;
+  ngtcp2_tstamp ts;
+  size_t pkt_count;
+  ngtcp2_path_storage ps;
+};
+
+static ngtcp2_tstamp timestamp(void)
+{
+  struct curltime ct = Curl_now();
```

Comment:
`Curl_now()` isn't guaranteed to always go forwards, in some situations it can jump back. For most modern platforms it is monotonic, however.

Reply:
The addition is a move, since the function is needed earlier in the code.

What is the alternative? Any recommendation?

Reply:
Unfortunately guaranteed monotonic clock adds platform dependency. There is no fully portable and widely available way to get truly monotonic clock.

Traditionally `clock_gettime(CLOCK_MONOTONIC, ..)` had been considered to be only increasing time stamp. However, for example macOS `CLOCK_MONOTONIC` isn't always monotonic and can actually jump back.

Depending on what is needed `CLOCK_MONOTONIC_RAW` might be suitable. This of course isn't available generally everywhere.